### DISCUSSION TITLE
Five Nights at Freddy's

### DISCUSSION THREAD

**Discussion by ghost** [skyline-emu/skyline#636] [Game Requests]
Right now, the 7 (2D) FNaF games on the Switch are:

Five Nights at Freddy's
Five Nights at Freddy's 2
Five Nights at Freddy's 3
Five Nights at Freddy's 4
Five Nights at Freddy's: Sister Location
Freddy Fazbear's Pizzeria Simulator
Ultimate Custom Night

Right now, all 7 of these games render upside-down on Ryujinx, which is interesting. I feel these are very simple games and that they should probably be tested with the current build even, since it's possible they might already function to some degree

Anyway, that's my request

**Comment by PixelyIon** [comment]
It seems like all the games run on the same engine (`Clickteam Fusion 2.5`), there's generally two reason for games rendering upside down on the Switch:
* Games use bottom-left (OpenGL) viewport origin — There's a load of flipping involved that needs to be emulated correctly for this
* Games flip the image to compensate for the Switch's internal display being placed flipped 180 degrees in the body — You can generally test this by toggling between docked/handheld

There's a lot of ways to flip the image, if you don't catch and implement some of the flips then the image will appear to be rendered incorrectly.
